allure-cucumber4-jvm/src/main/java/io/qameta/allure/cucumber4jvm/AllureCucumber4Jvm.java

```diff
@@ -110,13 +114,17 @@ public void setEventPublisher(final EventPublisher publisher) {
 
         publisher.registerHandlerFor(TestStepStarted.class, stepStartedHandler);
         publisher.registerHandlerFor(TestStepFinished.class, stepFinishedHandler);
+
+        publisher.registerHandlerFor(WriteEvent.class, writeEventHandler);
+        publisher.registerHandlerFor(EmbedEvent.class, embedEventHandler);
     }
 
     /*
     Event Handlers
      */
 
     private void handleFeatureStartedHandler(final TestSourceRead event) {
+        Allure.setLifecycle(lifecycle);
```

Comment:
what is this line for?

Reply:
Right, I was going to ask for your assistance here
When you call `Allure.addAttachment("Text output", event.text);` it creates a new lifecycle that is ignored in your mocked `runFeature` method.
Adding `Allure.setLifecycle(lifecycle);` helped but I am not sure if this a good solution.

Reply:
in fact the problem is that a new test lifecycle created during tests. So we need to set it to `Allure`. You can wrap `runtime.run()` method call (https://github.com/allure-framework/allure-java/blob/master/allure-cucumber4-jvm/src/test/java/io/qameta/allure/cucumber4jvm/AllureCucumber4JvmTest.java#L724) in test like we did it for TestNG https://github.com/allure-framework/allure-java/blob/master/allure-testng/src/test/java/io/qameta/allure/testng/AllureTestNgTest.java#L1114-L1124

Reply:
or you can use `getLifecycle().addAttachment(name, TEXT_PLAIN, TXT_EXTENSION, content.getBytes(StandardCharsets.UTF_8));` API directly (instead of using `Allure.addAttachment`)

Reply:
Alright, thanks. I'll look into it.